### PR TITLE
Update pot.erl to support sha256 and not use deprecated :crypto.hmac

### PR DESCRIPTION
**Pull request by franc**
https://github.com/yuce/pot/issues/29

support sha256

### CODE REVIEW

**Comment by franc** [comment]
#29

**Comment by nalundgaard** [comment]
This looks reasonable. I would like to wait on https://github.com/yuce/pot/pull/31 to be merged so that you can rebase on it and we can have passing CI record on all supported OTP versions, if that's OK!

**Comment by franc** [comment]
sure - happy to rebase once #31 is merged

**Comment by nalundgaard** [comment]
Well, the broken travis check stopped, but it doesn't seem to trigger the GitHub Action flow. that's really weird!

File: src/pot.erl [hunk 1]
```diff
@@ -90,8 +90,8 @@ hotp(Secret, IntervalsNo, Opts) ->
     TokenLength = proplists:get_value(token_length, Opts, 6),
     Key = pot_base32:decode(Secret),
     Msg = <<IntervalsNo:8/big-unsigned-integer-unit:8>>,
-    Digest = crypto:hmac(DigestMethod, Key, Msg),
-    <<_:19/binary, Ob:8>> = Digest,
+    Digest = crypto:mac(hmac, DigestMethod, Key, Msg),
```
Comment (nalundgaard) [inline src/pot.erl]: OK, looks like this doesn't work with OTP 20/21, and I would like to keep that compatibility. I would suggest abstracting this to a local function and using the new `OTP_RELEASE` flag like [this](https://github.com/erlcloud/erlcloud/blob/c02b1fe5dcdd5ec13702d958003881cce745edb5/src/erlcloud_util.erl#L28-L39). Note that you have to check whether it's defined AND whether it is >= 23 as in the example above.

Comment (franc) [reply src/pot.erl]: ok, i have added your suggestion.
my Erlang-fu is weak, but hopefully this works

**Review by nalundgaard — APPROVED** [review]
Awesome! LGTM and I see that this commit triggered a run of the GitHub Actions based checks! Thank you!

**Pull request merged**